kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Config.java

```diff
@@ -634,15 +634,23 @@ protected static ExecCredential getExecCredentialFromExecConfig(ExecConfig exec,
       List<ExecEnvVar> env = exec.getEnv();
       // TODO check behavior of tty & stdin
       ProcessBuilder pb = new ProcessBuilder(getAuthenticatorCommandFromExecConfig(exec, configFile, Utils.getSystemPathVariable()));
+      pb.redirectErrorStream(true);
       if (env != null) {
         Map<String, String> environment = pb.environment();
         env.forEach(var -> environment.put(var.getName(), var.getValue()));
       }
       Process p = pb.start();
+      StringBuilder sb = new StringBuilder();
+      try (BufferedReader in = new BufferedReader(new InputStreamReader(p.getInputStream()))) {
+        String line;
+        while ((line = in.readLine()) != null) {
+          sb.append(line);
+        }
+      }
       if (p.waitFor() != 0) {
-        LOGGER.warn(IOHelpers.readFully(p.getErrorStream()));
+        LOGGER.warn(sb.toString());
       }
-      ExecCredential ec = Serialization.unmarshal(p.getInputStream(), ExecCredential.class);
+      ExecCredential ec = Serialization.unmarshal(sb.toString(), ExecCredential.class);
```

Comment:
In this excerpt you are no longer considering the error stream.

Would something like this work with your solution?
```suggestion
      final String errorOutput = IOHelpers.readFully(p.getErrorStream());
      final String standardOutput = IOHelpers.readFully(p.getInputStream());
      if (p.waitFor() != 0) {
        LOGGER.warn(errorOutput);
      }
      ExecCredential ec = Serialization.unmarshal(standardOutput, ExecCredential.class);
```

Reply:
I am redirecting the error stream to outputstream with the following code
pb.redirectErrorStream(true);
and if the command exit code is other than 0 then reading the outputstream as error log, so we don't need to read ErrorStream anymore

Reply:
Oh, sorry I missed that line.
However, I still don't see why we removed `IOHelpers#readFully` method in favor of the read stream boilerplate code.

Reply:
@manusa because we have redirected all output to outputstream of the process and that will be read with the following code.

try (BufferedReader in = new BufferedReader(new InputStreamReader(p.getInputStream()))) {
        String line;
        while ((line = in.readLine()) != null) {
          sb.append(line);
        }
      }

and in case of failure **sb** variable will have the error messages
In case of error sb variable will have the error output

Reply:
Yes, but can't seem to understand why that stream can't be read using
```
final String output = IOHelpers.readFully(p.getInputStream());
```
which in essence encapsulate that expanded code (or should).

Reply:
I could read the output with that command however that helper method was not closing the anything. If you want I could change the 
`try (BufferedReader in = new BufferedReader(new InputStreamReader(p.getInputStream()))) {
        String line;
        while ((line = in.readLine()) != null) {
          sb.append(line);
        }
      }`

with `final String output = IOHelpers.readFully(p.getInputStream());`

Reply:
This should take care of closing the stream and avoid extra boilerplate code:
```java
String output;
try (InputStream is = p.getInputStream()) {
  output = IOHelpers.readFully(is);
}
```

My main concern is that I wouldn't like to keep same read stream logic lying around several parts of the codebase. It would be hard then to maintain buffer sizes or other stream reading configurations.
IOHelpers also requires a refactor. Besides, in case we want to delegate this to some library (e.g. commons-io), we'll only need to change this in a single place.

Reply:
It is updated with IOHelpers

Reply:
Great, thx!